### DISCUSSION TITLE
Feature request: boolean configuration option to pre-highlight "Yes" on `confirm-close-surface` pop-up

### DISCUSSION THREAD

**Discussion by thyttan** [ghostty-org/ghostty#2911] [Feature Requests, Ideas] [answered]
First, thanks for the beta invite and for creating ghostty!

Before reading [CONTRIBUTING.md](https://github.com/ghostty-org/ghostty/blob/main/CONTRIBUTING.md) I wrote this out in issue #2910, which I then closed.

# Feature request

- Add a boolean configuration option to pre-highlight the "Yes" button on the `confirm-close-surface` pop-up (name suggestion: `confirm-close-surface-highlight-yes`).
- ~~set the option to `true` by default (a little more risky than `false`, but it makes more sense to me).~~ Edit: Argued against on  https://github.com/ghostty-org/ghostty/discussions/2911#discussioncomment-12256239, which makes sense I guess.

# Rationale

The pop-up is in my mind a good idea. E.g. as a reminder for not accidentally leaving swap files around when exiting neovim without closing buffers in a nice way.

But coming from the ptyxis terminal emulator I experience some small extra friction when interacting with the pop-up in ghostty. In ptyxis the "Close" button is highlighted as soon as it's displayed. So closing the window is just a press of the enter key away. That feels like a good balance to me. Currently on ghostty I will either need to use the mouse to click "Yes", or do `<tab>` then `<enter>` on the keyboard.

Admittedly a small thing - but would be nice! Feel free to close the issue if it doesn't make sense to implement it.

# My system

- ghostty built from source (commit 49f105cd27ecb7afd58d57a2df2ef6ae6f2e1765) with `zig build -p $HOME/.local -Doptimize=ReleaseFast`
- Fedora Linux 41 (Workstation edition)
- Gnome Shell 47.1

**Comment by xzbdmw** [comment]
See https://github.com/ghostty-org/ghostty/discussions/3817#discussioncomment-11687840, but you are on linux, I believe there are something similar conventional keybinds.

**Comment by pluiedev** [comment, marked as answer]
The [GNOME Human Interface Guidelines](https://developer.gnome.org/hig/patterns/feedback/dialogs.html) explicitly recommend *against* making the close button the default, as users should be presented with the cancel button first, have time to think about their decision, and then click on said decision (emphasis mine):

> Always ensure that the cancel button appears first, before the affirmative button. In left-to-right locales, this is on the left. **This button order ensures that users become aware of, and are reminded of, the ability to cancel prior to encountering the affirmative button.**

The return key *especially* should never be used for destructive actions like closing a surface/window/tab/etc. (again emphasis mine):

> Assign the return key to activate the affirmative button. **However, this should not be done if its action is irreversible, destructive or otherwise inconvenient to the user.** If there is no appropriate button to designate as the default button, do not set one. 

It might sound a bit dogmatic, but this is exactly what human interface guidelines are designed to prevent: users making a mistake by accident without thinking through something that could be dangerous. Software should prevent people from making that class of mistakes. That friction is intentional: it avoids mistakes.

**Reply by thyttan** [reply]
Thanks for elaborating! I still think the option `confirm-close-surface-highlight-yes` would be nice but set to `false` by default then.

**Reply by pluiedev** [reply]
I mean, that's essentially offering the user a big footgun if they get careless with it. In that case then might as well turn `confirm-close-surface` to false entirely. Better to make users know exactly what they're in for, rather than offer a sense of false security.

**Reply by thyttan** [reply]
Sure. As stated this is not a big deal to me either way.
